test/integration/expression-tests/format/implicit-assert/test.json

```diff
@@ -0,0 +1,14 @@
+{
+  "expression": ["number", ["get", "p"]],
+  "propertySpec": {
+    "type": "formatted"
+  },
+  "expected": {
+    "compiled": {
+        "result": "error",
+        "errors": [
+          {"key": "", "error": "Expected formatted but found number instead."}
```

Comment:
This seems like a rather unhelpful error message, considering that most `"text-field"` values won't be actually using the `"format"` expression. Should we special case this type checking error for `formatted` or punt on this for a later error message improvement project?

Reply:
🤔 I think defer? The way I kind of think about it is: it's confusing because people expect to be able to return a 'string', and have it implicitly-coerce to 'formatted' without them having to even know 'formatted' exists. So you'd kind of like it to say something like `Expected formatted or string but found number instead.` -- but that's still actually confusing because (1) things other than 'formatted' or 'string' can still successfully coerce to 'formatted', (2) even 'number' can coerce to 'formatted' - the problem is that the assertion disables the implicit coercion.